src/main/java/it/pagopa/ecommerce/commons/domain/v1/TransactionUserCanceled.java

```diff
@@ -24,17 +24,17 @@
 @Getter
 public final class TransactionUserCanceled extends BaseTransaction implements Transaction {
 
-    TransactionUserCanceledEvent transactionUserCanceledEvent;
+    TransactionClosedEvent transactionClosedEvent;
 
     /**
      * Primary constructor
      *
-     * @param baseTransaction              the base transaction
-     * @param transactionUserCanceledEvent the transaction expired event
+     * @param baseTransaction        the base transaction
+     * @param transactionClosedEvent the transaction closed event
      */
     public TransactionUserCanceled(
             BaseTransaction baseTransaction,
```

Comment:
Fixed in https://github.com/pagopa/pagopa-ecommerce-commons/pull/48/commits/128e42ccc5dc56aa6a04e6c2c90665e311520e67
Since TransctionUserCanceledEvent has no data I made it extends directly BaseTransactionWithPaymentToken. Can this approach work? Or I have to made a base class with no additional data?

Reply:
Never mind, since I'm updating all others aggregate hierarchy also this one will be fixed with the same approach